app/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source.rb

```diff
@@ -33,4 +36,49 @@ def sync
     update!(:status => "error", :last_updated_on => Time.zone.now, :last_update_error => error)
     raise error
   end
+
+  private
+
+  # Find Terraform Templates(dir) in the git repo.
+  # Iterate throught git repo worktree, and collate all terraform template dir's (dirs with .tf or .tf.json files).
+  #
+  # Returns [hashmap] of template dirs and files within it.
+  def find_templates_in_git_repo()
+    template_dirs = {}
+
+    # checkout files to temp dir, we need for prasing input/output vars
+    git_checkout_tempdir = Dir.mktmpdir("terraform-git")
```

Comment:
@putmanoj is anything deleting this directory?  I think you need an `ensure` block at the end of this method to delete this otherwise they'll start to pile up in `/tmp` until a reboot.